Library/ViewModels/CommentsViewModel.swift

```diff
@@ -106,43 +145,31 @@ public final class CommentsViewModel: CommentsViewModelType,
       }
     )
 
-    let commentsAndProject = Signal.combineLatest(comments, initialProject)
+    let optimisticPostComment = Signal.combineLatest(
+      initialProject,
+      currentUser.skipNil()
+    ).takePairWhen(self.postCommentButtonTappedProperty.signal.skipNil())
+      .map(unpack)
+      .map(Comment.optimisticComment)
+
+    self.postCommentSubmitted = optimisticPostComment.ignoreValues()
+
+    let commentsWithOptimisticPostComment = Signal
+      .combineLatest(comments, optimisticPostComment)
+      .map { loadedComments, optimisticComment -> [Comment] in
+        var comments = loadedComments
+        comments.insert(optimisticComment, at: 0)
+        return comments
+      }
+
+    let commentsAndProject = Signal.combineLatest(
```

Comment:
A few questions: 
1. Once the users' comment successfully gets posted, how do we remove the optimistic comment or do we not care about that? Which leads me to wonder...
2. After the comment has succeeded, refreshing the comments might cause an issue where `optimisticComment` is always added to the comments (even with the just posted comment in there already) because `optimisticComment`'s latest value will be emitted (ie. `combineLatest` will send its last value, which is comment already in `comments`).

Reply:
I haven't tested it, so let me know if I'm misunderstanding somehow :)

Reply:
Also, probably not part of this PR, but it would be helpful to add a `TODO: // What happens if comment posted fails`, which ties into a ticket regarding that comment state @moderateepheezy

Reply:
This line of code `let commentsAndProject = Signal.combineLatest(
      Signal.merge(comments, commentsWithOptimisticPostComment),
      initialProject
    )`
ensures that when `comments` is emitted it would replace the datasource comments

Reply:
Yeah I added a `// TODO:` on `ln 96` in `CommentsViewModel.swift`